### ISSUE TITLE
How to tell which frames are loaded by SequenceReader?

### ISSUE THREAD

**Issue by twang9** [NVIDIA/DALI#1603]
Hi, everyone. 

Just a quick question, how to get the names/paths of the images/frames loaded by SequenceReader? In my project I need to identify the loaded frames to load their corresponding ground truth. But in doc I can't find a way to do this. 
Meanwhile, my case is not like #1445 that the ground truth is image that can be accessed by FileReader, but some typical xml file. 

Thank you!

**Comment by JanuszL** [comment]
Hi,
Currently, it is not possible and `SequenceReader` is rather rudimentary as it was developed as an auxiliary operator to verify the VideoLoader functionality. If your ground truth format is the same as a training data (number of frames, etc) you can have to readers working in parallel with the same speed in the directories with the same structure (the same sub-folder, etc).
Have you considered to convert your data to video?

**Comment by twang9** [comment]
@JanuszL Hi Januszal, thanks for your reply. My data can be converted to video. Initially I think decoding frames may faster than videos. So, if I need to do with video, do I need to generate the corresponding file_list.txt with the format:
video_file.mp4 start_frame end_frame label
Then, how do I deal with the label part if my labels are xml files per frame? Thanks.

**Comment by JanuszL** [comment]
`SequenceReader` is not tightly coupled with the host decoder so the decoding itself could be a bottleneck.
You can check the nightly build and API for the [`VideoReader`](https://docs.nvidia.com/deeplearning/sdk/dali-master-branch-user-guide/docs/supported_ops.html#nvidia.dali.ops.VideoReader).
The file format, if I remember correctly is:
```
video_file.mp4 label start_frame end_frame
```
If you assign a unique label to each file and enable frame number as an additional output using `enable_frame_num` you can easily map the data to corresponding xml (label and frame number).

**Comment by twang9** [comment]
Thank you I will look into it.

**Comment by twang9** [comment]
Hi @JanuszL . After I switch to VideoReader following the #1599 and [this](https://github.com/NVIDIA/DALI/blob/master/docs/examples/video/superres_pytorch/dataloading/dataloaders.py) with the latest nightly build (0.18), the iterator showed a strange move. It just seems paused after about 9 sequences (paused after 2 iterations with batch_size=4, and after 1 with batch_size=8, etc). 
So I just wonder if you happen to know what may cause this and how to solve this? Thank you.

BTW: I tried on my two machines, cuda 10.1 and 10.2 with different driver versions, different cards, still the same.

**Comment by twang9** [comment]
After I tried with different batch_size, sequence_length combinations, it seems the pipeline stop working after providing a certain number of frames. Larger batch_size or sequence_length -> smaller iteration num, smaller batch_size or sequence_length -> larger iteration num.

**Comment by JanuszL** [comment]
@tcwang0509 - is it possible to get the video and the pipeline configuration that causes this hang? Is this video a variable frame rate kind one (DALI doesn't support it, there is some heuristic to detect and raise an error but it is not 100% accurate).
@a-sansanwal - anything comes to your mind?

**Comment by a-sansanwal** [comment]
Hi @twang9 @JanuszL 
I believe the issue you mention most likely could be related #1592(which should be merged soon) or related to variable frame rate heuristic.
I can take a look at it if you can upload the videos. It will help improve DALI.

**Comment by JanuszL** [comment]
Please check the nightly build and see if that works for you.

**Comment by JanuszL** [comment]
0.19 is out and should address this. Please reopen if it doesn't work.

**Comment by tonysy** [comment]
@JanuszL Hi, I am working on a video project, while I have extracted all frames into jpg images. Each time, I generate a sequence of frames path(with specific temporal stride and start/end index). I would like to use the sequencereader to read the sequence images. I wonder how to use the squencereader by giving a sequence of image paths directly. Thanks

**Comment by JanuszL** [comment]
@tonysy - the `SequenceReader` can only read files from folders. You cannot specify image paths directly. As it was mentioned already the `SequenceReader` functionality is rather rudimentary. Still, we would be more than happy to accept any PR extending its fucnitonality.
